os/gtimer/gtimer_timer.go

```diff
@@ -233,7 +233,7 @@ func (t *Timer) binSearchIndex(n int64) (index int, result int) {
 	mid := 0
 	cmp := -2
 	for min <= max {
-		mid = int((min + max) / 2)
+		mid = min + int((max-min)/2)
```

Comment:
I do not see any bug here, would you please have a little bit explanation about this change?